---
bool ready
bool reporting
bool active
